gensim/models/hdpmodel.py

```diff
@@ -80,65 +122,125 @@ def lda_e_step(doc_word_ids, doc_word_counts, alpha, beta, max_iter=100):
         expElogtheta = np.exp(Elogtheta)
         phinorm = np.dot(expElogtheta, betad) + 1e-100
         meanchange = np.mean(abs(gamma - lastgamma))
-        if (meanchange < meanchangethresh):
+        if meanchange < meanchangethresh:
             break
 
     likelihood = np.sum(counts * np.log(phinorm))
     likelihood += np.sum((alpha - gamma) * Elogtheta)
     likelihood += np.sum(gammaln(gamma) - gammaln(alpha))
     likelihood += gammaln(np.sum(alpha)) - gammaln(np.sum(gamma))
 
-    return (likelihood, gamma)
+    return likelihood, gamma
 
 
 class SuffStats(object):
+    """Stores suff stats for document(s)."""
+
     def __init__(self, T, Wt, Dt):
+        """
+
+        Parameters
+        ----------
+        T : int
+            Top level truncation level.
+        Wt : int
+            Length of words in the documents.
+        Dt : int
+            Chunk size.
+
+        """
         self.m_chunksize = Dt
         self.m_var_sticks_ss = np.zeros(T)
         self.m_var_beta_ss = np.zeros((T, Wt))
 
     def set_zero(self):
+        """Fill the sticks and beta array with 0 scalar value."""
         self.m_var_sticks_ss.fill(0.0)
         self.m_var_beta_ss.fill(0.0)
 
 
 class HdpModel(interfaces.TransformationABC, basemodel.BaseTopicModel):
-    """
-    The constructor estimates Hierachical Dirichlet Process model parameters based
-    on a training corpus:
-
-    >>> hdp = HdpModel(corpus, id2word)
-
-    You can infer topic distributions on new, unseen documents with
-
-    >>> doc_hdp = hdp[doc_bow]
-
-    Inference on new documents is based on the approximately LDA-equivalent topics.
-
-    To print 20 topics with top 10 most probable words
-
-    >>> hdp.print_topics(num_topics=20, num_words=10)
-
-    Model persistency is achieved through its `load`/`save` methods.
+    """`Hierarchical Dirichlet Process model <http://jmlr.csail.mit.edu/proceedings/papers/v15/wang11a/wang11a.pdf>`_
```

Comment:
It will be really nice if you describe here, how the model works in general (so many parameters, so small understanding, why this needed here).

Reply:
@menshikh-iv Should the explanation be using the parameters name or just an overview of how the model functions ?

Reply:
@gyanesh-m simple principle - more information is better (imagine, than your documentation for person who knows nothing about topic modeling)

Reply:
@menshikh-iv Sorry for the delay but I got caught up in some work yesterday so couldn't do much. I am going through the paper to get a better understanding and I will try to finish this today.

Reply:
@gyanesh-m great, don't rush, the better description more important than speed, we'll wait

Reply:
@menshikh-iv Hi, I have written the explanation for hdp model . You can find it [here](https://docs.google.com/document/d/1aKIOErwEO1kyykBuTeu7NiH9K7FGTcENuHQz7NCT5vc/edit?usp=sharing). I have enabled adding comments, so that you can give suggestions easily.

Reply:
@gyanesh-m great description! You definitely should add it to `hdpmodel.py` :+1:

Reply:
@gyanesh-m :+1:
about your proposal - I'll answer you soon (I'm on the business trip, sorry for waiting)

Reply:
@menshikh-iv Ok, np. Also, the headings mentioned in the google doc should be formatted as bold or as a separate heading ?

Reply:
@menshikh-iv I have formatted the headings in bold for now since I couldn't find the solution to make them as separate sub heading.

Reply:
yes, that's a good idea